pkg/artifactory/resource_artifactory_remote_repository_test.go

```diff
@@ -67,8 +67,54 @@ func TestKeyHasSpecialCharsFails(t *testing.T) {
 	})
 }
 
+/*type TestCheckFunction func(string ...) resource.TestCheckFunc
+
+type FunStruct struct {
+	fieldName         string
+	testCheckFunction TestCheckFunction
+}*/
+//resource.TestCheckResourceAttr(resourceName, "repo_layout_ref", getDefaultLocalRepoLayoutRef(repoType)),
+
+/*func getTestCheckFuncs() func(int, int) int {
+	return func(x, y int) int {
+		return x * y
+	}
+}*/
+
+/*
+	funStructs := []FunStruct {
+		FunStruct{
+			fieldName: "repo_layout_ref",
+			testCheckFunction: func(resourceName string, fieldName string, repoType string) resource.TestCheckFunc {
+				return resource.TestCheckResourceAttr(resourceName, "repo_layout_ref", getDefaultLocalRepoLayoutRef(repoType))
+			},
+		},
+	}
+	fmt.Println(funStructs)
+*/
+
 func TestAccRemoteDockerRepository(t *testing.T) {
-	_, testCase := mkNewRemoteTestCase("docker", t, map[string]interface{}{
+	const packageType = "docker"
+	_, testCase := mkNewRemoteTestCase(packageType, t, map[string]interface{}{
+		"external_dependencies_enabled":  true,
+		"enable_token_authentication":    true,
+		"block_pushing_schema1":          true,
+		"priority_resolution":            false,
+		"external_dependencies_patterns": []interface{}{"**/hub.docker.io/**", "**/bintray.jfrog.io/**"},
+		"missed_cache_period_seconds":    1800, // https://github.com/jfrog/terraform-provider-artifactory/issues/225
+		"content_synchronisation": map[string]interface{}{
+			"enabled":                         false, // even when set to true, it seems to come back as false on the wire
+			"statistics_enabled":              true,
+			"properties_enabled":              true,
+			"source_origin_absence_detection": true,
+		},
+	})
+	resource.Test(t, testCase)
+}
+
+func TestAccRemoteDockerRepositoryWithAdditionalCheckFunctions(t *testing.T) {
```

Comment:
How is this test case different from `TestAccRemoteDockerRepository`? Same for the rest in this file.

Reply:
`TestAccRemoteDockerRepositoryWithAdditionalCheckFunctions` is invoking separate function `mkRemoteTestCaseWithAdditionalCheckFunctions`. Where we are passing additional `addCheckFunctions`. Currently `mkNewRemoteTestCase` not providing the capability of additional check functions.